src/main/java/org/spongepowered/api/entity/ai/task/builtin/VanillaAITask.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.entity.ai.task.builtin;
+
+import org.spongepowered.api.entity.ai.task.AITask;
+import org.spongepowered.api.entity.living.Agent;
+
+/**
+ * Represents an {@link AITask} that exists in vanilla minecraft.
+ *
+ * @param <O> The type of agent
+ */
+public interface VanillaAITask<O extends Agent> extends AITask<O> {
+
+    /**
+     * Get the mutex bits of an AI task in vanilla. The mutex bit is not
+     * supposed to be used, and might not exist in some different
+     * implementations. See {@link AITask#canRunConcurrentWith(AITask)}, which
+     * achieves the function of mutex bits.
+     *
+     * @deprecated Platform specific, might not be available
+     * @return The mutex bits of this vanilla AI task.
+     */
+    @Deprecated
+    int getMutexBits();
```

Comment:
After sleeping on this, I think the mutex stuff + Vanilla parent interface should move to a separate PR as finding a correct solution to it shouldn't hold up the rest of this PR (which is pretty good).

Since they both target bleeding, breaking the API in the next PR isn't a problem.